sklearn/tree/_tree.pyx

```diff
@@ -243,7 +244,7 @@ cdef class DepthFirstTreeBuilder(TreeBuilder):
                                          split.threshold, impurity, n_node_samples,
                                          weighted_n_node_samples)
 
-                if node_id == SIZE_MAX:
+                if <size_t> node_id == SIZE_MAX:
```

Comment:
Thanks, @OmarManzoor. Can you instead change the type of `node_id` at its declaration?